cluster-autoscaler/integration/framework.go

```diff
@@ -103,9 +107,9 @@ func (driver *Driver) adjustNodeGroups() error {
 
 	for index, machineDeployment := range machineDeployments.Items {
 		scaleDownMachineDeployment := machineDeployment.DeepCopy()
-		if index > 0 && machineDeployment.Spec.Replicas != 0 {
+		if (index == 0 || index > 1) && machineDeployment.Spec.Replicas != 0 {
```

Comment:
instead search by `machineDeployment` name , so that in future we can add more worker groups also easily